dvc/cache/base.py

```diff
@@ -614,18 +615,25 @@ def get_files_number(self, path_info, hash_info, filter_info):
         )
 
     def _to_dict(self, dir_info):
-        return {
-            entry[self.tree.PARAM_RELPATH]: entry[self.tree.PARAM_CHECKSUM]
-            for entry in dir_info
-        }
+        info = {}
+        for entry in dir_info:
+            relpath = None
+            hash_info = None
+            for key, value in entry.items():
+                if key == self.tree.PARAM_RELPATH:
+                    relpath = value
+                else:
+                    hash_info = HashInfo(key, value)
+            info[relpath] = hash_info
+        return info
```

Comment:
This looks a bit ugly but dir_info may actually contain hashes with a different type than `cache.tree.PARAM_CHECKSUM` - i.e. dir_info contains S3 etag entries, but we are working in local cache (w/md5 `PARAM_CHECKSUM`)